Listener stack smash detected and crashes

**Describe the bug**
If I run fogros in 22.04 / docker container with the current humble branch, the listener crashes with 
```
[INFO] [launch]: Default logging verbosity is set to INFO
action added
[INFO] [talker-1]: process started with pid [4048]
[talker-1] 1650477066.806642 [0]     talker: config: //CycloneDDS/Domain/General: 'NetworkInterfaceAddress': deprecated element (file:///home/ubuntu/cyclonedds.xml line 6)
[listener-1] *** stack smashing detected ***: terminated
[talker-1] [WARN] [1650477067.375230800] [minimal_publisher]: Publishing: "Hello World: 0"
[ERROR] [listener-1]: process has died [pid 214, exit code -6, cmd '/home/root/fog_ws/install/lib/fogros2_examples/listener --ros-args'].
[talker-1] [WARN] [1650477067.861732385] [minimal_publisher]: Publishing: "Hello World: 1"
[talker-1] [WARN] [1650477068.361687046] [minimal_publisher]: Publishing: "Hello World: 2"
[talker-1] [WARN] [1650477068.861811733] [minimal_publisher]: Publishing: "Hello World: 3"
[talker-1] [WARN] [1650477069.361692310] [minimal_publisher]: Publishing: "Hello World: 4"
```

**To Reproduce**
Steps to reproduce the behavior:
(just follow the docker setup process in README) 

**Screenshots**

<img width="886" alt="Screen Shot 2022-04-20 at 10 55 12 AM" src="https://user-images.githubusercontent.com/25167951/164292928-a2c45a4a-03bc-4d65-994a-50696214560f.png">

--- a/Dockerfile
+++ b/Dockerfile
@@ -1,13 +1,13 @@
 ARG DISTRO=rolling
-FROM ros:$DISTRO
+FROM ubuntu:jammy
 
 ARG DISTRO
 
-RUN apt update && apt install -y curl gnupg2 lsb-release && \
+RUN apt update && apt install -y curl gnupg2 lsb-release sudo && \
     curl -sSL https://raw.githubusercontent.com/ros/rosdistro/master/ros.key -o /usr/share/keyrings/ros-archive-keyring.gpg
     
 RUN echo "deb [arch=$(dpkg --print-architecture) signed-by=/usr/share/keyrings/ros-archive-keyring.gpg] http://packages.ros.org/ros2/ubuntu jammy main" | sudo tee /etc/apt/sources.list.d/ros2.list > /dev/null
-RUN apt install -y ros-rolling-desktop
+RUN apt update && DEBIAN_FRONTEND=noninteractive apt install -y ros-rolling-desktop
 
 RUN apt update && sudo apt install -y \
   build-essential \

--- a/Dockerfile
+++ b/Dockerfile
@@ -3,6 +3,12 @@ FROM ros:$DISTRO
 
 ARG DISTRO
 
+RUN apt update && apt install -y curl gnupg2 lsb-release && \
+    curl -sSL https://raw.githubusercontent.com/ros/rosdistro/master/ros.key -o /usr/share/keyrings/ros-archive-keyring.gpg
+    
+RUN echo "deb [arch=$(dpkg --print-architecture) signed-by=/usr/share/keyrings/ros-archive-keyring.gpg] http://packages.ros.org/ros2/ubuntu jammy main" | sudo tee /etc/apt/sources.list.d/ros2.list > /dev/null
+RUN apt install -y ros-rolling-desktop
+
 RUN apt update && sudo apt install -y \
   build-essential \
   cmake \

